crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -123,6 +123,73 @@ impl AcirContext {
         let is_equal_witness = self.acir_ir.is_equal(&lhs_expr, &rhs_expr);
         self.add_data(AcirVarData::Witness(is_equal_witness))
     }
+
+    /// Returns an `AcirVar` that is the XOR result of `lhs` & `rhs`.
+    pub(crate) fn xor_var(&mut self, lhs: AcirVar, rhs: AcirVar) -> Result<AcirVar, AcirGenError> {
+        let lhs_bit_size =
+            self.variables_to_bit_sizes.get(&lhs).expect("ICE: XOR applied to field type");
```

Comment:
Note: Not blocking -- Can we figure out why this is a compiler error and not a user error, ie which of the previous compilers ensure that this should not happen?

Reply:
Can create an issue as I think we may have done this in other places in this file too

Reply:
The type system should prevent using xor on field types I believe so it should not be possible for users to write code to trigger this.

Reply:
Will add a suggestion to specify that its the type system